src/dev.c

```diff
@@ -12,6 +12,75 @@
 
 static int nonce = 1234; // The only cryptographically secure nonce
 
+
+static void fido_dev_set_extension_flags(fido_dev_t *dev, const fido_cbor_info_t *info) {
+    // TODO
+    /*char * const   *ptr = fido_cbor_info_extensions_ptr(info);
+    size_t          len = fido_cbor_info_extensions_len(info);
+
+    for (size_t i = 0; i < len; i++)
+        if (strcmp(ptr[i], "credProtect") == 0)
+            dev->flags |= FIDO_DEV_CRED_PROT;
+    */
+}
+
+static void fido_dev_set_option_flags(fido_dev_t *dev, const fido_cbor_info_t *info) {
+    // TODO
+    /*
+    char * const   *ptr = fido_cbor_info_options_name_ptr(info);
+    const bool     *val = fido_cbor_info_options_value_ptr(info);
+    size_t          len = fido_cbor_info_options_len(info);
+
+    for (size_t i = 0; i < len; i++) {
+        if (strcmp(ptr[i], "clientPin") == 0) {
+            dev->flags |= val[i] ?
+                          FIDO_DEV_PIN_SET : FIDO_DEV_PIN_UNSET;
+        } else if (strcmp(ptr[i], "credMgmt") == 0 ||
+                   strcmp(ptr[i], "credentialMgmtPreview") == 0) {
+            if (val[i]) {
+                dev->flags |= FIDO_DEV_CREDMAN;
+            }
+        } else if (strcmp(ptr[i], "uv") == 0) {
+            dev->flags |= val[i] ?
+                          FIDO_DEV_UV_SET : FIDO_DEV_UV_UNSET;
+        } else if (strcmp(ptr[i], "pinUvAuthToken") == 0) {
+            if (val[i]) {
+                dev->flags |= FIDO_DEV_TOKEN_PERMS;
+            }
+        }
+    }
+    */
+}
+
+static void fido_dev_set_protocol_flags(fido_dev_t *dev, const fido_cbor_info_t *info) {
+    // TODO
+    /*
+    const uint8_t  *ptr = fido_cbor_info_protocols_ptr(info);
```

Comment:
Those could be implemented in this PR.

Reply:
Done

Reply:
@felix-gohla Please check if you are fine with the new implementation.